src/psbt/mod.rs

```diff
@@ -37,15 +48,39 @@ impl PsbtUtils for Psbt {
             None
         }
     }
+
+    fn fee_amount(&self) -> Option<u64> {
+        let tx = &self.unsigned_tx;
+        let utxos: Option<Vec<TxOut>> = (0..tx.input.len()).map(|i| self.get_utxo_for(i)).collect();
+
+        utxos.map(|inputs| {
+            let input_amount = inputs.iter().fold(0, |acc, i| acc + i.value);
```

Comment:
I think you can use [`.sum()`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.sum) here instead of manually implementing it with `.fold()`.

Reply:
ugh 🤦 thanks  fixed!